pinot-core/src/main/java/org/apache/pinot/core/operator/filter/predicate/PredicateUtils.java

```diff
@@ -145,23 +145,26 @@ public static IntSet getDictIdSet(BaseInPredicate inPredicate, Dictionary dictio
         }
         break;
       case STRING:
-        if (queryContext == null || values.size() <= Integer.parseInt(queryContext.getQueryOptions()
-            .getOrDefault(CommonConstants.Broker.Request.QueryOptionKey.IN_PREDICATE_SORT_THRESHOLD,
-                CommonConstants.Broker.Request.QueryOptionValue.DEFAULT_IN_PREDICATE_SORT_THRESHOLD))) {
+        if (queryContext == null || Boolean.parseBoolean(
```

Comment:
If else block is always better, then we can remove this?

Reply:
Just want to give it an option to turn it off in case sorting is dominating the latency (very unlikely though, but who knows)